mlir/test/Dialect/Vector/vector-transfer-collapse-inner-most-dims.mlir

```diff
@@ -169,3 +169,27 @@ func.func @non_unit_strides(%arg0: memref<512x16x1xf32, strided<[8192, 16, 4], o
 // The inner most unit dims can not be dropped if the strides are not ones.
 // CHECK:     func.func @non_unit_strides
 // CHECK-NOT:   memref.subview
+
+// -----
+
+func.func @contiguous_outer_dyn_inner_most_view2(%a: index, %b: index, %memref: memref<?x?x8x1xf32>) -> vector<8x1xf32> {
```

Comment:
It's not obvious what makes this case unique - perhaps just update `contiguous_outer_dyn_inner_most_view` instead? Alternatively, 
* `contiguous_outer_dyn_inner_most_view2` -> `contiguous_outer_dyn_inner_most_view_2_dyn_dims` (to highlight makes this unique),
* move near `contiguous_outer_dyn_inner_most_view` so that it's easy to contrast both cases (and find tests for similar scenarios).

This is really just a nit, feel free to ignore.

Reply:
(this was the test case that failed before, because the input memref was not strided, I think)

Reply:
Sure, but that context is gone once this is merged (unless you check the commit msg). I still think that it would be helpful to highlight what differentiates this case.

Reply:
SGTM, I think updating the test covers the needs, and it is a more robust test.